src/v1/device_manager_client.ts

```diff
@@ -2105,6 +2153,17 @@ export class DeviceManagerClient {
               res.on('end', () => {
                 const deviceStatesRes: protos.google.cloud.iot.v1.IListDeviceStatesResponse =
                   JSON.parse(data);
+                if (isBinaryDataFormat()) {
+                  deviceStatesRes.deviceStates?.forEach((element, index) => {
```

Comment:
Same as above ^. Use Array.prototype.map instead of mutating with forEach. It's the idiomatic way of transforming data in an array in JS